static/js/components/tiles/map_tile.tsx

```diff
@@ -256,175 +286,222 @@ export function getReplacementStrings(
+    const facetIds = layer.variable.facetId ? [layer.variable.facetId] : null;
+    const placeStatPromise: Promise<PointApiResponse> = getPointWithin(
+      props.apiRoot,
+      layer.enclosedPlaceType,
+      layer.parentPlace,
+      [layer.variable.statVar],
+      dataDate,
+      [],
+      facetIds
     );
-  } catch (error) {
-    removeSpinner(props.id);
-    return null;
+    const populationPromise: Promise<SeriesApiResponse> = props.statVarSpec
+      .denom
+      ? getSeriesWithin(
+          props.apiRoot,
+          layer.parentPlace,
+          layer.enclosedPlaceType,
+          [layer.variable.denom]
+        )
+      : Promise.resolve(null);
+    const parentPlacesPromise = props.parentPlaces
+      ? Promise.resolve(props.parentPlaces)
+      : axios
+          .get(
+            `${props.apiRoot || ""}/api/place/parent?dcid=${layer.parentPlace}`
+          )
+          .then((resp) => resp.data);
+    try {
+      const [geoJson, placeStat, population, parentPlaces, borderGeoJsonData] =
+        await Promise.all([
+          geoJsonPromise,
+          placeStatPromise,
+          populationPromise,
+          parentPlacesPromise,
+          borderGeoJsonPromise,
+        ]);
+      // Only draw borders for containing places without 'wall to wall' coverage
+      const borderGeoJson = shouldShowBorder(layer.enclosedPlaceType)
+        ? borderGeoJsonData
+        : undefined;
+      const rawData = {
+        borderGeoJson,
+        enclosedPlaceType: layer.enclosedPlaceType,
+        geoJson,
+        parentPlaces,
+        place,
+        placeStat,
+        population,
+      };
+      rawDataArray.push(rawData);
+    } catch (error) {
+      removeSpinner(props.id);
+      return null;
+    }
   }
+  return rawToChart(rawDataArray, layers[0].variable, props);
 };
 
 function rawToChart(
-  rawData: RawData,
+  rawDataArray: RawData[],
   statVarSpec: StatVarSpec,
-  place: NamedTypedPlace,
-  enclosedPlaceType: string,
   props: MapTilePropType
 ): MapChartData {
-  const metadataMap = rawData.placeStat.facets || {};
-  const placeStat = rawData.placeStat.data[statVarSpec.statVar] || {};
-
   const dataValues = {};
+  const dates: Set<string> = new Set();
   const metadata = {};
+  const placeData = [];
   const sources: Set<string> = new Set();
-  const dates: Set<string> = new Set();
-  if (_.isEmpty(rawData.geoJson)) {
-    return;
-  }
-  const { unit, scaling } = getStatFormat(statVarSpec, rawData.placeStat);
-  for (const geoFeature of rawData.geoJson.features) {
-    const placeDcid = geoFeature.properties.geoDcid;
-    const placeChartData = getPlaceChartData(
-      placeStat,
-      placeDcid,
-      false /* set isPerCapita as false here so that we can calculate per capita the same way as all the tiles */,
-      {},
-      metadataMap
-    );
-    if (_.isEmpty(placeChartData)) {
-      continue;
+  const units = {};
+  let isUsaPlace = true;
```

Comment:
add comment that this is only true if all layers are a us place

Reply:
Added comment.